infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -63,33 +92,22 @@
     private final ConcurrentHashMap<UUID, Long> streamTailToGlobalTailMap = new ConcurrentHashMap<>();
 
     /**
-     * A cache of conflict parameter modification timestamps.
+     * A cache that records recent (stream, conflict parameter) modification
+     * timestamps (global addresses).
      *
-     * A "wildcard" {@link SequencerServer::maxConflictWildcard} representing all the entries which were evicted from the cache holds
-     * theier maximal modification time.
+     * And a "wildcard" representing all the entries which were evicted from the cache
+     * holds their maximal modification time.
      */
     private final long maxConflictCacheSize = 10_000;
     private long maxConflictWildcard = -1L;
-    private final Cache<Integer, Long> conflictToGlobalTailCache = Caffeine.newBuilder()
+    private final Cache<AbstractMap.SimpleEntry<Integer,Integer>, Long>
```

Comment:
this is the best way to get a pair in Java . The pair serves as a two-part key, (stream-id, conflict-param). 

The streadID is a UUID, which may consume too much memory, especially since there will be many repetitions (same-stread-ID, different-conflict-param). Hashing into Integer seems safe enough, I don't expect stream-IDs to collide that easily.

We may think about a two-tier cache in the future, one cache per stream-ID. It really depends on the workload: if there are few streams and many keys in each one, it would save on memory. If there are many streams and few keys in each, it would be wasteful. For now, I think that using a pair of integers as a unique conflict-key is probably good enough.